### PR TITLE
chore: fixes potential cross site scripting

### PR DESCRIPTION
**Pull request by naveensrinivasan**
## Description

- fixes  `G203: The used method does not auto-escape HTML. This can potentially lead to 'Cross-site Scripting' vulnerabilities in case the attacker controls the input. (gosec)`

## Related Issue

Fixes #

Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: src/internal/packager/sbom/viewer.go [hunk 1]
```diff
@@ -44,8 +47,8 @@ func (b *Builder) createSBOMHTML(filename string, goTemplate string, jsonData []
 	}{
 		ThemeCSS:  b.loadFileCSS("theme.css"),
 		ViewerCSS: b.loadFileCSS("styles.css"),
-		List:      template.JS(b.jsonList),
-		Data:      template.JS(jsonData),
+		List:      template.JS(safeJsonList),
+		Data:      template.JS(safeJsonData),
```
Comment (Racer159) [inline src/internal/packager/sbom/viewer.go:51]:
```suggestion
		List:      template.JS(safeJSONList),
		Data:      template.JS(safeJSONData),
```
linting

**Review by Racer159 — CHANGES_REQUESTED** [review]

File: src/internal/packager/sbom/viewer.go [hunk 2]
```diff
@@ -31,6 +31,9 @@ func (b *Builder) createSBOMHTML(filename string, goTemplate string, jsonData []
 
 	defer sbomViewerFile.Close()
 
+	safeJsonData := template.JSEscapeString(string(jsonData))
+	safeJsonList := template.JSEscapeString(string(b.jsonList))
```
[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply src/internal/packager/sbom/viewer.go]: Thanks. I took care of this.

File: src/internal/packager/sbom/viewer.go (hunk 1 above)
Comment (naveensrinivasan) [reply src/internal/packager/sbom/viewer.go]: Thanks. I took care of this.

File: src/internal/packager/sbom/viewer.go [hunk 3]
```diff
@@ -44,8 +47,8 @@ func (b *Builder) createSBOMHTML(filename string, goTemplate string, jsonData []
 	}{
 		ThemeCSS:  b.loadFileCSS("theme.css"),
 		ViewerCSS: b.loadFileCSS("styles.css"),
-		List:      template.JS(b.jsonList),
-		Data:      template.JS(jsonData),
+		List:      template.JS(safeJSONList),
+		Data:      template.JS(safeJSONData),
```
Comment (Racer159) [inline src/internal/packager/sbom/viewer.go:51]: Should have checked earlier, but this actually breaks the way we are including these values into the sbom html files:

image

(note the escaped values)

image

simple repro:

```
zarf package create examples/dos-games/
zarf package inspect zarf-package-dos-games-amd64-1.0.0.tar.zst -s
```

**Review by Racer159 — CHANGES_REQUESTED** [review]

**Comment by Racer159** [comment]
After more thought and discussion, closing this since the threat surface is lower (would require modifying syft json data and there aren't any creds to steal on the site given it is a static HTML page).  Fixing this would likely require more effort than it is worth at this juncture.

**Pull request closed without merging**